dynamo-zip-store/src/main/scala/dynamo_zip_store/DynamoZipStore.scala

```diff
@@ -0,0 +1,56 @@
+package dynamo_zip_store
+
+
+import cats.effect.IO
+import fs2.Stream
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient
+import software.amazon.awssdk.services.dynamodb.model.{AttributeValue, GetItemRequest, PutItemRequest}
+import zip_partitioner.FileArchive
+
+import scala.jdk.CollectionConverters.MapHasAsJava
+
+object DynamoZipStore {
+
+
+
+  def uploadFilesToDynamoDB(client: DynamoDbClient, fileArchives: List[FileArchive], tableName: String): IO[Unit] = {
+
+    fileArchives.foreach { fileArchive =>
+      val fileName =fileArchive.name
+      val fileBytes = fileArchive.compressedData
+
+      val item = Map(
+        "fileName" -> AttributeValue.builder().s(fileName).build(),
+        "data" -> AttributeValue.builder().s(fileBytes).build()
+      ).asJava
+
+      val request = PutItemRequest.builder()
+        .tableName(tableName)
+        .item(item)
+        .build()
+
+      client.putItem(request)
```

Comment:
async client will give you a completable future, use IO.fromCompletableFuture when you switch